patterns/state/ships/src/main/java/com/scottlogic/dp/state/Main.java

```
@@ -1,21 +1,16 @@
 package com.scottlogic.dp.state;
 
-import com.scottlogic.dp.state.ships.FishingShip;
+import com.scottlogic.dp.state.ships.Ship;
 
 public class Main {
     public static void main(String... args) {
-        FishingShip yourBarge = new FishingShip();
-        yourBarge.stop();
-        yourBarge.sail();
-        yourBarge.stop();
+        Ship yourBarge = new Ship();
+        yourBarge.startSailing();
+        yourBarge.stopSailing();
+        yourBarge.lowerAnchor();
         yourBarge.startFishing();
-        yourBarge.anchor();
-        yourBarge.startFishing();
-        yourBarge.stop();
-        yourBarge.raiseAnchor();
         yourBarge.stopFishing();
-        yourBarge.sail();
         yourBarge.raiseAnchor();
-        yourBarge.sail();
+        yourBarge.startSailing();
     }
```

You've removed the invalid transitions because they don't make sense, that's fair enough.
As we spoke before, the output is not the same, that is fine.
However, have you considered that before, the invalid transition allowed the program to carry on but now invalid transition stops the execution?
Is it desirable?
What would you do about that?

---

I think it could be clearer that it is actually something you aren't allowed to do by throwing an error rather than just printing a line maybe? And maybe you could catch the error somewhere else if it needed to be handled so that the program could continue?

---

Yep. There's nothing wrong with it, I am just pointing it out :)